### DISCUSSION TITLE
What are the use cases for Assumptions.assumingThat(...)?

### DISCUSSION THREAD

**Discussion by Marcono1234** [junit-team/junit-framework#2553] [Q&A] [answered]
The `Assumptions` class has two [`assumingThat(...)`](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assumptions.html#assumingThat(boolean,org.junit.jupiter.api.function.Executable)) methods.

What is the intended use case for them?
It appears they just act like a regular `if` statement except that they allow throwing checked exceptions without having to declare them. Though personally I don't think adding `throws Exception` to the test method signature (which has no negative side-effects) is so cumbersome that it needs an `assumingThat` method as solution.

Additionally it is slightly irritating that these `assumingThat` methods are in the `Assumptions` class which describes itself as:
> In direct contrast to failed assertions, failed assumptions do not result in a test failure; rather, a failed assumption results in a test being aborted.

The `assumingThat` methods don't actually abort the test.

Or is it about the intention they convey, indicating that a conditional execution in a test method is intentionally based on the test environment?

**Comment by sbrannen** [comment, marked as answer]
> Or is it about the intention they convey, indicating that a conditional execution in a test method is intentionally based on the test environment?

Yes, that's what they are for.

They are effectively nothing more than glorified if-statements that serve as a DSL for the reader of  the test.

**Reply by Marcono1234** [reply]
Thank you for the answer. Do you think it would be good to explicitly state that, unlike the other assumption methods, these do not cause the test to abort (and instead can even cause a test failure when an exception is thrown)?

**Reply by sbrannen** [reply]
The [Javadoc](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assumptions.html#assumingThat(java.util.function.BooleanSupplier,org.junit.jupiter.api.function.Executable)) for each `assumingThat(...)` method already makes such statements.

**Reply by Marcono1234** [reply]
Hmm, ok it does say:
> If the assumption is invalid, this method does nothing. 

I was expecting something explicit like:
> Unlike the other assumption methods, this method will not abort the test. If the assumption is invalid, this method does nothing. If the assumption is valid and the executable throws an exception, it will be treated like a regular test _failure_.

If you want I can create a pull request for that.

**Reply by sbrannen** [reply]
Sure, feel free to open a PR for that documentation improvement.
